File: KutuphaneWeb/library-automation-web/src/types/loan.ts
export default interface Loan {
    id?: number;
    accountId?: number;
    accountUserName?: string;
    accountFirstName?: string;
    accountLastName?: string;
    loanLines: LoanLine[];
    loanDate: string;
    dueDate: string;
    returnDate?: string;
    status?: "OnLoan" | "Returned" | "Overdue" | "Canceled";
    displayStatus?: string;
    fineAmount?: number;
}

export interface LoanLine {
    id?: number;
    loanId?: number;
    bookId: number;
    bookTitle?: string;
    availableCopies?: number;
    bookAuthor?: string;
    bookISBN?: string;
    bookImageUrl?: string;
    quantity: number;
}

File: KutuphaneAPI/Entities/Models/Loan.cs
namespace Entities.Models
{
    public class Loan
    {
        public int Id { get; set; }
        public string AccountId { get; set; } = null!;
        public Account? Account { get; set; }
        public ICollection<LoanLine>? LoanLines { get; set; }
        public DateTime LoanDate { get; set; } = DateTime.UtcNow;
        public DateTime DueDate { get; set; }
        public DateTime? ReturnDate { get; set; }
        public LoanStatus Status { get; set; } = LoanStatus.OnLoan;
        public int? PenaltyId { get; set; }
        public decimal? FineAmount { get; set; }
    }

    public enum LoanStatus
    {
        OnLoan,
        Returned,
        Overdue,
        Canceled
    }
}
